plugins/uefi-capsule/fu-uefi-capsule-device.c

```diff
@@ -570,6 +597,12 @@ fu_uefi_capsule_device_probe(FuDevice *device, GError **error)
 		fu_device_add_private_flag(device,
 					   FU_UEFI_CAPSULE_DEVICE_FLAG_NO_CAPSULE_HEADER_FIXUP);
 
+	/* if the bootloader doesn't know how to install capsules provided by fwupd,
+	 * enable fwupd-efi */
+	if (!fu_uefi_capsule_device_bootloader_supports_fwupd(device)) {
+		fu_device_add_private_flag(device, FU_UEFI_CAPSULE_DEVICE_FLAG_USE_FWUPD_EFI);
+	}
```

Comment:
can we move this to `fu-uefi-capsule-plugin.c` please? In the case there are 12 ESRT entries we'd do 12x `fu_efivars_get_data()` -- e.g. I think it would slot in `fu_uefi_capsule_plugin_coldplug_device()` quite nicely. Otherwise patch LGTM.

Reply:
Makes sense, moved to `fu_uefi_capsule_plugin_coldplug_device` as suggested.